ufl/algorithms/analysis.py

```diff
@@ -198,19 +200,20 @@ def extract_base_form_operators(a):
     return sorted_by_count(extract_type(a, BaseFormOperator))
 
 
-def extract_arguments_and_coefficients(a):
-    """Build two sorted lists of all arguments and coefficients in a.
+def extract_terminals_with_domain(a):
+    """Build three sorted lists of all arguments, coefficients, and geometric quantities in `a`.
 
-    This function is faster than extract_arguments + extract_coefficients
-    for large forms, and has more validation built in.
+    This function is faster than extracting each type of terminal
+    separately for large forms, and has more validation built in.
 
     Args:
         a: A BaseForm, Integral or Expr
     """
```

Comment:
Could we add typing of input/output here?

Reply:
Done.